react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -53,58 +63,72 @@ export class QRCode extends Component {
       username: '',
       qrcode: '',
       activity: 'not selected',
-      activities: []
-    }
+      activities: [],
+    };
 
     this.handleVideo = this.handleVideo.bind(this);
 
     this.changeActivity = this.changeActivity.bind(this);
   }
 
-  handleVideo = (e) => {
+  handleVideo = e => {
     let newState = {};
-    newState['login'] = this.state.login+1;
+    newState['login'] = this.state.login + 1;
     this.setState(newState);
   };
 
-  changeActivity = (newActivity) => {
+  changeActivity = newActivity => {
     this.setState({
-      activity: newActivity
+      activity: newActivity,
     });
 
     const visitInfo = {
       hash: this.state.hash,
-      activity: newActivity
-    }
+      activity: newActivity,
+    };
 
     return fetch('/postActivity', {
-      method: "POST",
-      body: JSON.stringify(visitInfo)
-    }).then(
-      this.props.history.push('/visitor/end')
-    );
+      method: 'POST',
+      body: JSON.stringify(visitInfo),
+    })
+      .then(this.props.history.push('/visitor/end'))
+      .catch(error => {
+        console.log('ERROR HAPPENING AT FETCH /postActivity', error);
```

Comment:
if something goes wrong posting the activity to the server, what do you want to do? Rollback the entire log in? Prompt the user to log in again? May be worth considering, and put that functionality here.

Reply:
@ZooeyMiller , thank you so much for such a comprehensive review! please, have a look at the changes: line 93 forwards to login page. What do you think is better: to add a new component similar to "qrerror" that says "Sorry, we cannot record your purpose of visit, please log in again" or just forward to login page without explaining anything?

Reply:
I think you definitely need to explain that the login was unsuccessful, as far as I understand it's an important part of the login process that the activity is tracked right? So a login should be considered unsuccessful if an activity isn't tracked? 

Rather than creating a new component, you could try passing options to `qrerror` depending on where the error occurred? And it could display a different error message depending on where the error happened?

Whatever the specific implementation is, I think it's important that the user isn't just redirected to login with no information, they need to be told that a problem occurred.